web/files/files.go

```diff
@@ -307,13 +307,23 @@ func FileCopyHandler(c echo.Context) error {
 		return WrapVfsError(err)
 	}
 
+	copyName := c.QueryParam("Name")
+	if copyName == "" || filepath.Ext(copyName) != filepath.Ext(olddoc.DocName) {
+		copyName = fileCopyName(inst, olddoc.DocName)
+	}
+
+	newdoc := vfs.CreateFileDocCopy(olddoc, copyName)
+
+	if newDirID := c.QueryParam("dir_id"); newDirID != "" {
+		newdoc.DirID = newDirID
+		newdoc.ResetFullpath()
+	}
+
 	err = checkPerm(c, permission.POST, nil, olddoc)
```

Comment:
We should check the permission on `newdoc`. If an app has a permission on a folder, it should not be able to create a copy of a file in this filter at the root of the Cozy.

Reply:
Ah good catch. That was the intention but I forgot to replace `olddoc` with `newdoc` after moving the check.